### PR TITLE
nat: describe ui page and netmap

### PR DESCRIPTION
**Pull request by gsanchietti**
This is a work in progress.

Things to do:
- [x] make sure addresses in the examples are correct
- [x] verify the general example for netmap
- [x] explain the source netmap and add an example
- [x] explain the destination netmap and an example

Card: https://trello.com/c/tavYHA5b/283-nat-rules-source-and-destination-netmap

### CODE REVIEW

**Comment by cotosso** [comment]
The doc should be ok, please check the last part if there are any errors in the commands

File: nat.rst [hunk 1]
```diff
@@ -26,4 +32,129 @@ are translated into a single public IP address when communicating with external
...
+
+From CLI create a rule::
+
+ uci set netmap.r1=rule
+ uci set netmap.r1.name=source_nat
+ uci set netmap.r1.dest=10.2.2.0/24
+ uci set netmap.r1.map_from=192.168.1.0/24
+ uci set netmap.r1.map_to=10.1.1.0/24
+
+
```
Comment (gsanchietti) [inline nat.rst:126]:
```suggestion
```

File: nat.rst [hunk 2]
```diff
@@ -26,4 +32,129 @@ are translated into a single public IP address when communicating with external
...
+The masquerade rule masks all outgoing traffic with the IP address of the firewall's outbound interface.
+Traffic from internal hosts to the Internet are automatically masqueraded by the firewall.
+Masquerade can also be used to mask traffic originating from a remote network (e.g., VPN) with the firewall's IP to avoid any routing issues.
+
+**Example** You need to reach a host on the local network (routed) from the VPN network (e.g. 192.168.7.0/24), but the host doesn't have a configured gateway or has a different gateway than the firewall.
+
+**Problem** The host cannot reach the local device due to the lack of a gateway.
+
+**Solution** Create a NAT rule with masquerade action for the traffic coming from the VPN Network.This masks the traffic from the VPN network (192.168.7.0/24) to the local network with the firewall's IP of the destination interface.
```
Comment (gsanchietti) [inline nat.rst:63]:
```suggestion
**Solution** Create a NAT rule with masquerade action for the traffic coming from the VPN Network. This masks the traffic from the VPN network (192.168.7.0/24) to the local network with the firewall's IP of the destination interface.
```

File: nat.rst [hunk 3]
```diff
@@ -26,4 +32,129 @@ are translated into a single public IP address when communicating with external
...
+Disable NAT (no-NAT)
+====================
+
+Disabling NAT (no-NAT) allows you to bypass the NAT process for specific traffic. 
+This is particularly useful when it comes to avoiding WAN masquerading for specific destinations.
+
+**Example** Your firewall is connected to a router that, in addition to allowing internet access, also enables reaching private networks through CDN connections or IPsec tunnels. 
+In order to reach private remote networks, traffic from local network must come out with its original ip address (no masquerade rewriting).
+
+**Problem** The router tunnel policies only allow traffic between the NethSecurity local network and the destination networks, but all traffic comes out of nethsecurity with the masked IP (NethSecurity WAN IP).
```
Comment (gsanchietti) [inline nat.rst:80]:
```suggestion
**Problem** The router tunnel policies only allow traffic between the NethSecurity local network and the destination networks, but all traffic comes out of the firewall with the masked IP (NethSecurity WAN IP).
```

File: nat.rst [hunk 4]
```diff
@@ -26,4 +32,129 @@ are translated into a single public IP address when communicating with external
...
+.. _netmap-section:
+
+Netmap
+======
+
+Netmap is a NAT technique that offers 1:1 network-wide translation without changing the individual host addresses.
+This means it could map an entire private network (e.g., 192.168.1.0/24) to a another network (e.g., 10.5.6.0/24) at once,
+eliminating the need to manually configure individual NAT rules for each device.
+
+**Example** 2 firewalls, A and B holding a VPN tunnel between Networks A and B, local and remote networks are overlapping (192.168.1.0/24), so this makes it impossible to route traffic between them.
```
Comment (gsanchietti) [inline nat.rst:96]:
```suggestion
**Example** 2 firewalls, F1 and F1 holding a VPN tunnel between networks A and B, local and remote networks are overlapping (192.168.1.0/24), so this makes it impossible to route traffic between them. 
```

File: nat.rst [hunk 5]
```diff
@@ -26,4 +32,129 @@ are translated into a single public IP address when communicating with external
...
+Translate A and B networks onto two alternative networks can solve the problem so that there are no overlapping networks.
+
+Let's use this translation scheme.
+
+* Network A: 192.168.1.0/24 -> is translated to -> Network ALT_A: 10.1.1.0/24
+* Network B: 192.168.1.0/24 -> is translated to -> Network ALT_B: 10.2.2.0/24
+
+A host in network A trying to reach a host in network B must not contact the real IP but its translated network (only the last octet remains the same). 
+For example, the host 192.168.1.10 from the network A wanting to reach 192.168.0.20 in network B must contact the IP 10.2.2.20 instead.
+Before the request exits firewall A (FW-A), the source of the packet will be rewritten by FW-A to the ALT_IP 10.1.1.10 to eliminate every routing issue on network B. The same process will occur for the returning packets.
```
Comment (gsanchietti) [inline nat.rst:106]:
```suggestion
Before the request exits firewall  FW-A, the source of the packet will be rewritten by FW-A to the ALT_IP 10.1.1.10 to eliminate every routing issue on network B. The same process will occur for the returning packets.
```

File: nat.rst [hunk 6]
```diff
@@ -26,4 +32,129 @@ are translated into a single public IP address when communicating with external
...
+
+
+**Solution** The problem can be solved by using netmap to translate the traffic to a different private network. This allows the traffic to be routed correctly.
+
+
+**Result** The traffic between networks A and B will be routed correctly.
+
+Source Netmap
+-------------
+The "source netmap" allows us to determine how the source should change when traffic is directed towards a specific destination. E.g., destination network 10.2.2.0/24, source network: 192.168.0.0/24, natted source network: 10.1.1.0/24.
```
Comment (gsanchietti) [inline nat.rst:116]:
```suggestion
-------------

The "source netmap" allows us to determine how the source should change when traffic is directed towards a specific destination. E.g., destination network 10.2.2.0/24, source network: 192.168.0.0/24, natted source network: 10.1.1.0/24.
```

File: nat.rst [hunk 7]
```diff
@@ -26,4 +32,129 @@ are translated into a single public IP address when communicating with external
...
+From CLI create a rule::
+
+ uci set netmap.r2=rule
+ uci set netmap.r2.name=dest_nat
+ uci set netmap.r2.src=10.2.2.0/24
+ uci set netmap.r2.map_from=10.1.1.0/24
+ uci set netmap.r2.map_to=192.168.1.0/24
+
+
+
```
Comment (gsanchietti) [inline nat.rst:151]:
```suggestion
```

**Pull request merged**